gitops/sre/.tekton/gateway-deployment.yaml

```diff
@@ -0,0 +1,61 @@
+apiVersion: tekton.dev/v1beta1
+kind: PipelineRun
+metadata:
+  name: gateway-deployment
+  annotations:
+    pipelinesascode.tekton.dev/on-target-branch: "[main]"
+    pipelinesascode.tekton.dev/on-event: "[push]"
+    pipelinesascode.tekton.dev/task: "git-clone"
+spec:
+  params:
+    - name: repo_url
+      value: "{{repo_url}}"
+    - name: revision
+      value: "{{revision}}"
+  pipelineSpec:
+    params:
+      - name: repo_url
+      - name: revision
+    workspaces:
+      - name: source
+    tasks:
+      - name: fetch-repository
+        taskRef:
+          name: git-clone
+        workspaces:
+          - name: output
+            workspace: source
+        params:
+          - name: depth
+            value: "500"
+          - name: url
+            value: $(params.repo_url)
+          - name: revision
+            value: $(params.revision)
+      - name: kcp-register
+        runAfter:
+          - fetch-repository
+        workspaces:
+          - name: source
+            workspace: source
+        taskSpec:
+          workspaces:
+            - name: source
+          steps:
+            - name: gateway-deployment
+              image: ghcr.io/openshift-pipelines/gateway-deployment:main
+              workingDir: $(workspaces.source.path)
+              env:
+                - name: DATA_DIR
+                  value: $(workspaces.source.path)/gitops/sre
```

Comment:
We have set this value as  $(workspaces.source.path) for the other two images. And in the script, it is indeed referenced like this. So, could you please change the value for the cluster-setup.yaml and kcp-registration.yaml too?
files=($(ls $DATA_DIR/credentials/kubeconfig/compute))

Reply:
We do not need the gitops/sre as folders 'credentials' and 'environment' are placed at the root level.
Please ignore my previous comment. And in this file, we need to remove '/gitops/sre'.

Reply:
> We have set this value as $(workspaces.source.path) for the other two images. And in the script, it is indeed referenced like this. So, could you please change the value for the cluster-setup.yaml and kcp-registration.yaml too?

This PR is focused on the gateway. It is unfortunate that changes made by PR #86 broke other things.

> And in this file, we need to remove '/gitops/sre'.

I disagree. The location of the credentials directory is under gitops/sre/credentials, which is exactly what you get if you fork this repository. Other approaches of creating a gitops repository, which may not require a gitops/sre directory are acceptable but then the persons creating such a repository can adjust the environment variable to their custom structure.

The documentation, which should have come with PR #86  is unfortunately missing. That's the reason I created [issue #105 ](https://github.com/openshift-pipelines/pipelines-service/issues/105). It is unfortunate that it has not been worked on.

Reply:
- Currently, both the cluster-setup and kcp-registrar PR have workingDir: $(workspaces.source.path). 
- And even in the register.sh script for kcp-registrar, we are using $DATA_DIR/credentials/kubeconfig where DATA_DIR will point to wherever credentials and environment are present.

https://github.com/openshift-pipelines/pipelines-service/blob/main/gitops/sre/.tekton/kcp-registration.yaml#L47
https://github.com/openshift-pipelines/pipelines-service/blob/main/gitops/sre/.tekton/pln-svc-registration.yaml#L47

Reply:
changed it for the sake of consistency with the other PipelineRuns